PhysObjectExtractor/src/GenParticleAnalyzer.cc

```diff
@@ -0,0 +1,214 @@
+// -*- C++ -*-
+//
+// Package:    GenParticleAnalyzer
+// Class:      GenParticleAnalyzer
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+//classes to extract GenParticle information
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
+
+//classes to save data
+#include "TTree.h"
+#include "TFile.h"
+#include<vector>
+
+//
+// class declaration
+//
+
+class GenParticleAnalyzer : public edm::EDAnalyzer {
+   public:
+      explicit GenParticleAnalyzer(const edm::ParameterSet&);
+      ~GenParticleAnalyzer();
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      virtual void beginJob() ;
+      virtual void analyze(const edm::Event&, const edm::EventSetup&);
+      virtual void endJob() ;
+      virtual void beginRun(edm::Run const&, edm::EventSetup const&);
+      virtual void endRun(edm::Run const&, edm::EventSetup const&);
+      virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&);
+      virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+     std::vector<std::string>  particle;
+
+      // ----------member data ---------------------------
+
+      TTree *mtree;
+
+      std::vector<int> GenPart_status;
+      std::vector<float> GenPart_pt;
+      std::vector<float> GenPart_eta;
+      std::vector<float> GenPart_mass;
+      std::vector<int> GenPart_pdgId;
+      std::vector<float> GenPart_phi;
+      std::vector<float> GenPart_px;
+      std::vector<float> GenPart_py;
+      std::vector<float> GenPart_pz;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+
+GenParticleAnalyzer::GenParticleAnalyzer(const edm::ParameterSet& iConfig):
+particle(iConfig.getParameter<std::vector<std::string> >("input_particle"))
```

Comment:
See, here you read the vstring from the configuration, but I don't think you ever use it.
I suggest to recover this string vector, loop over it, and get pdgids and status numbers from it.